File: ExternalAPI/client/src/main/java/external/api/ExternalAPIClient.java
package external.api;

import org.springframework.cloud.netflix.feign.FeignClient;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;

/**
 * Created by loicd on 20/05/2017.
 */
@FeignClient("external-api")
public interface ExternalAPIClient {

    @RequestMapping(method = RequestMethod.GET, path = "/soundcloud/sourceurl")
    public ResponseEntity<String> getSoundcloudSourceurlFromRessource(@RequestParam("ressource") String ressource);

}

File: ExternalAPI-client/src/scala/com/ldz/external/api/ExternalAPIClient.scala
package com.ldz.external.api

import com.ldz.external.api.model.ExternalMusicDTO
import org.springframework.cloud.netflix.feign.FeignClient
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.{RequestMapping, RequestMethod, RequestParam}

/**
  * Created by Loic on 14/06/2017.
  */
@FeignClient("external-api")
trait ExternalAPIClient {

  @RequestMapping(method = Array(RequestMethod.GET), path = Array("/soundcloud/sourceurl"))
  def getSoundcloudSourceurlFromRessource(@RequestParam("ressource") ressource: String): ResponseEntity[ExternalMusicDTO]

}
